packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -4375,5 +4386,532 @@ describe("Teal Opcodes", function () {
+      const stack = new Stack<StackElem>();
+
+      it("should push 0/1 depending on value of two byte arrays for ByteGreaterThanEqualTo", function () {
+        // A is not >= B
+        stack.push(parsing.stringToBytes("A"));
+        stack.push(parsing.stringToBytes("B"));
+        let op = new ByteGreaterThanEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 0n);
+
+        // B is >= A
+        stack.push(parsing.stringToBytes("B"));
+        stack.push(parsing.stringToBytes("A"));
+        op = new ByteGreaterThanEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 1n);
+
+        // A is >= A
+        stack.push(parsing.stringToBytes("A"));
+        stack.push(parsing.stringToBytes("A"));
+        op = new ByteGreaterThanEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 1n);
+      });
+    });
+
+    describe("ByteEqualTo", () => {
+      const stack = new Stack<StackElem>();
+
+      it("should push 0/1 depending on value of two byte arrays for ByteEqualTo", function () {
+        // A is not == B
+        stack.push(parsing.stringToBytes("A"));
+        stack.push(parsing.stringToBytes("B"));
+        let op = new ByteEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 0n);
+
+        // A == A
+        stack.push(parsing.stringToBytes("A"));
+        stack.push(parsing.stringToBytes("A"));
+        op = new ByteEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 1n);
+
+        // "" == ""
+        stack.push(parsing.stringToBytes(""));
+        stack.push(parsing.stringToBytes(""));
+        op = new ByteEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 1n);
+      });
+    });
+
+    describe("ByteNotEqualTo", () => {
+      const stack = new Stack<StackElem>();
+
+      it("should push 0/1 depending on value of two byte arrays for ByteNotEqualTo", function () {
+        // A is not == B
+        stack.push(parsing.stringToBytes("A"));
+        stack.push(parsing.stringToBytes("B"));
+        let op = new ByteNotEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 1n);
+
+        // A == A
+        stack.push(parsing.stringToBytes("A"));
+        stack.push(parsing.stringToBytes("A"));
+        op = new ByteNotEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 0n);
+
+        // "" !== ""
+        stack.push(parsing.stringToBytes(""));
+        stack.push(parsing.stringToBytes(""));
+        op = new ByteNotEqualTo([], 0);
+        op.execute(stack);
+        assert.deepEqual(stack.pop(), 0n);
+      });
+    });
+
+    describe("ByteBitwiseOr", () => {
+      const stack = new Stack<StackElem>();
+
+      it("should push OR of two byte arrays for ByteBitwiseOr", function () {
+        stack.push(hexToByte('0x11'));
+        stack.push(hexToByte('0x10'));
```

Comment:
is there a reason why are we using hex ? we could have used bigint.

Reply:
no specific reason, values are taken directly from go-algorand tests (which were in hex), so let's use hex here